src/lib/wasm/RaygunStore.ts

```diff
@@ -535,6 +539,77 @@ class RaygunStore {
         return messages
     }
 
+    /**
+     * Create a handler for attachment results that uploads the file to chat and updates pending message attachments
+     * TODO: verify it works as we dont have a way to upload files yet
+     */
+    private async createFileAttachHandler(conversationId: string, upload: wasm.AttachmentResult) {
+        let listener = {
+            [Symbol.asyncIterator]() {
+                return upload
+            },
+        }
+        let cancelled = false
+        for await (const value of listener) {
+            let event = parseJSValue(value)
+            log.info(`Handling file progress event: ${JSON.stringify(event)}`)
+            switch (event.type) {
+                case "AttachedProgress": {
+                    let locationKind = parseJSValue(event.values[0])
+                    // Only streams need progress update
+                    if (locationKind.type === "Stream") {
+                        let progress = parseJSValue(event.values[1])
+                        let file = progress.values["name"]
+                        ConversationStore.updatePendingMessages(conversationId, upload.get_message_id(), file, current => {
+                            if (current) {
+                                switch (progress.type) {
+                                    case "CurrentProgress": {
+                                        current.size = progress.values["current"]
+                                        current.total = progress.values["total"]
+                                        break
+                                    }
+                                    case "ProgressComplete": {
+                                        current.size = progress.values["total"]
+                                        current.total = progress.values["total"]
+                                        current.done = true
+                                        break
+                                    }
+                                    case "ProgressFailed": {
+                                        current.size = progress.values["last_size"]
+                                        current.error = progress.values["error"]
+                                        break
+                                    }
+                                }
+                                current
```

Comment:
This is line intended? I don't believe it does anything.

Reply:
my mistake. it shouldve returned